src/composables/use-search-type.ts

```diff
@@ -21,20 +23,28 @@ const icons = {
   [IMAGE]: imageIcon,
   [MODEL_3D]: model3dIcon,
 }
-const searchTypes = [...supportedSearchTypes]
 
 export default function useSearchType() {
   const activeType = computed(() => useSearchStore().searchType)
   const previousSearchType = ref(activeType.value)
-  const setActiveType = (searchType) => {
+  const featureFlagStore = useFeatureFlagStore()
+  const additionalTypes = computed(() =>
+    featureFlagStore.isOn('external_sources') ? ([MODEL_3D] as const) : []
+  )
+  const searchTypes = [...supportedSearchTypes]
+  const setActiveType = (
+    searchType: SupportedSearchType | typeof additionalTypes['value'][number]
+  ) => {
     if (previousSearchType.value === searchType) return
-    useSearchStore().setSearchType(searchType)
-    previousSearchType.value = searchType
+    useSearchStore().setSearchType(searchType as SupportedSearchType)
```

Comment:
I think we need to add a check for whether the search type is in `supportedSearchTypes` here, otherwise if for some reason we pass an additional type to the `useSearchStore.setSearchType`, it will throw an error.

Reply:
Ah, okay, so this function really should only accept SupportedSearchType? The additional types are just links and don't modify the store, is that right :thinking: